library/core/src/mem/mod.rs

```diff
@@ -1317,7 +1317,7 @@ impl<T> SizedTypeProperties for T {}
 /// ```
 #[cfg(not(bootstrap))]
 #[unstable(feature = "offset_of", issue = "106655")]
-#[allow_internal_unstable(builtin_syntax)]
+#[allow_internal_unstable(builtin_syntax, hint_must_use)]
 pub macro offset_of($Container:ty, $($fields:tt).+ $(,)?) {
-    builtin # offset_of($Container, $($fields).+)
+    crate::hint::must_use(builtin # offset_of($Container, $($fields).+))
```

Comment:
I think this needs a `$` at the front

Reply:
No it doesn't because this is a macros 2.0 macro, it is resolved at macro's definition.

Reply:
It is? That's cool.